rust/index/src/fulltext/types.rs

```diff
@@ -161,6 +162,40 @@ impl FullTextIndexWriter {
         Ok(())
     }
 
+    pub async fn prefetch_before_writing_to_blockfiles(
+        &self,
+        blockfile_provider: &BlockfileProvider,
+    ) -> Result<(), FullTextIndexError> {
+        // `token_instances` can be millions of items. Imagine there are 10k documents, each with 1000 tokens. There are then ~10000*1000*3=30M token instances.
+        // In comparison, the number of blocks required to store these is much, much smaller. Say (worst case) each token instance consumes 12 bytes of a block (trigram, offset ID, position) and a block is 8MB. Then a single block can store ~660k token instances.
+        // Given this, and assuming that token instances are relatively uniformly distributed across blocks, we can prefetch a very small percentage of the key space and the cache hit rate will still be at or near 100%.
+        const PERCENT_TO_PREFETCH: f32 = 0.0005;
```

Comment:
to give an idea of how much sampling impacts time:

0.05% took 1.3s to prefetch
50% took 2.5s to prefetch

(compacting scidocs in tilt, built in debug mode)

it's not a huge difference, but since this will be executing in an IO thread we should avoid doing lots of sync comp